Assets/Standard Assets/Characters/FirstPersonCharacter/Scripts/FirstPersonController.cs

```diff
@@ -47,6 +50,26 @@ public class FirstPersonController : MonoBehaviour
         [System.NonSerialized]
         public bool isStunned;
 
+        public GameObject FindClosestKeyLocation()
+        {
+            GameObject[] gos;
+            gos = GameObject.FindGameObjectsWithTag("KeyLocation");
+            GameObject closest = null;
+            float distance = Mathf.Infinity;
+            Vector3 position = transform.position;
+            foreach (GameObject go in gos)
+            {
+                Vector3 diff = go.transform.position - position;
+                float curDistance = diff.sqrMagnitude;
+                if (curDistance < distance)
+                {
+                    closest = go;
+                    distance = curDistance;
+                }
```

Comment:
Proper variable names would help to understand what is this. If the purpose of this whole function is to find the closest key location and return its object, then we may want to create some custom function to find the closest object with passing a tag, cause we use this operation quite a bit all over the app I see. And this function doesn't seem to be the best solution readability wise as well

Reply:
Will need to be remade anyway. Leave it for now